File: Self_Dividing_Numbers/solution.best.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function(left, right) {
  let ans = [];
  while (left <= right) {
    if (isSelfDividing(left)) ans.push(left);
    left++;
  }
  return ans;
};

function isSelfDividing (target) {
  let n = target;
  while (n > 0) {
    let d = n % 10;
    if (d === 0 || target % d !== 0) return false;
    n = Math.floor(n / 10);
  }
  return true;
}

File: Self_Dividing_Numbers/solution.best.py
class Solution:
  def selfDividingNumbers(self, left: int, right: int) -> List[int]:
    ans = []
    while left <= right:
      if self._isSelfDividing(left) == True: ans.append(left)
      left += 1
    return ans
  
  def _isSelfDividing(self, target: int):
    n = target
    while n > 0:
      d = n % 10
      if d == 0 or target %d != 0: return False
      n = math.floor(n / 10)
    return True
